src/WrappedMToken.sol

```diff
@@ -502,38 +541,29 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
 
     /**
      * @dev   Increments total earning supply by `amount_` tokens.
-     * @param amount_       The present amount of tokens to increment total earning supply by.
-     * @param currentIndex_ The current index used to compute the principal amount.
+     * @param amount_    The present amount of tokens to increment total earning supply by.
+     * @param principal_ The principal amount of tokens to increment total earning principal by.
      */
-    function _addTotalEarningSupply(uint240 amount_, uint128 currentIndex_) internal {
+    function _addTotalEarningSupply(uint240 amount_, uint112 principal_) internal {
         unchecked {
             // Increment the total earning supply and principal proportionally.
             totalEarningSupply += amount_;
-            principalOfTotalEarningSupply += IndexingMath.getPrincipalAmountRoundedUp(amount_, currentIndex_);
```

Comment:
why the was a round up in previous version? Bug? round down makes more sense

Reply:
Needed to round up because individual account principals were not stored and thus could not be accumulated in a fixed way, so `principalOfTotalEarningSupply` needed to be rounded up (inflated) to ensure excess was conservative (deflated).